ui/src/app/components/status-alert-banner.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+
+import {Button} from 'app/components/buttons';
+import {FlexColumn, FlexRow} from 'app/components/flex';
+import {ClrIcon} from 'app/components/icons';
+import {statusAlertApi} from 'app/services/swagger-fetch-clients';
+import colors, {colorWithWhiteness} from 'app/styles/colors';
+import {cookiesEnabled} from 'app/utils';
+
+const styles = {
+  alertBanner: {
+    backgroundColor: colorWithWhiteness(colors.highlight, .5),
+    color: colors.primary,
+    marginRight: '1rem',
+    marginTop: '1rem',
+    padding: '0.5rem',
+    width: '300px',
+    borderRadius: '0.5em',
+    position: 'absolute',
+    top: '0',
+    right: '0',
+    zIndex: '1',
+  }
+};
+
+const cookieKey = 'status-alert-banner-dismissed';
+
+export interface State {
+  statusAlertId: number;
+  title: string;
+  message: string;
+  link: string;
+  isVisible: boolean;
+}
+
+export class StatusAlertBanner extends React.Component<{}, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      statusAlertId: 0,
+      title: '',
+      message: '',
+      link: '',
+      isVisible: false
+    };
+  }
+
+  async componentDidMount() {
+    const statusAlerts = await statusAlertApi().getStatusAlert();
+    if (statusAlerts.length > 0) {
+      // There should be either one or zero alerts.
+      const statusAlert = statusAlerts[0];
+      this.setState({
+        statusAlertId: statusAlert.statusAlertId,
+        title: statusAlert.title,
+        message: statusAlert.message,
+        link: statusAlert.link
+      });
+      // This needs to happen separately so that the other state is already set in isVisible()
+      this.setState({
```

Comment:
I believe setState is asynchronous. So we can't depend on the previous setState having resolved. https://css-tricks.com/understanding-react-setstate/ 

If we are depending on state existing, we should use the updater function. Alternatively, we could do the styling in a ternary on the banner object, and then it will pull whatever is right at render time, as opposed to state.

Basically removing this state variable, and changing the style from
`style={this.getStyles(isVisible)}` => `style={{...styles.alertBanner, height: this.isVisible() ? '200px' : '0px'}}`, and removing the `getStyles` function.

(Has the added benefit of removing state, which is always good)

Reply:
Removing state is not inherently good in all cases. We shouldn't be forcing react components to rerender or change appearance, it should always be based off state and props. The Reactiest way to do this is probably binding a `hideStatusAlert` to the `NavBar` to pass as `onClick` to the `StatusAlertBanner`.